plutus-metatheory/src/VerifiedCompilation/UCaseOfCase.lagda.md

```diff
@@ -0,0 +1,100 @@
+```
+module VerifiedCompilation.UCaseOfCase where
+
+```
+
+## Imports
+
+```
+open import Data.Nat using (ℕ;zero;suc)
+open import Data.Empty using (⊥)
+open import Scoped using (ScopedTm; Weirdℕ)
+open import Untyped using (_⊢; case; builtin; _·_; force; `; ƛ; delay; con; constr; error)
+open import Untyped.CEK using (BApp; fullyAppliedBuiltin; BUILTIN; stepper; State; Stack)
+open import Evaluator.Base using (maxsteps)
+open import Builtin using (Builtin; ifThenElse)
+open import Data.List using (List; zip; [_])
+open import Utils as U using (Maybe;nothing;just;Either)
+open import RawU using (TmCon)
+open import Data.List.Relation.Unary.All using (All; []; _∷_)
+open import Data.Product using (_×_; proj₁; proj₂)
+open import Relation.Nullary using (Dec; yes; no)
+```
+
+## Translation Relation
+
+This compiler stage only applies to the very specific case where an `IfThenElse` builtin exists in a `case` expression.
+It moves the `IfThenElse` outside and creates two `case` expressions with each of the possible lists of cases. 
+
+This translation relation has a constructor for the specific case, and then inductive constructors for everything else
+to traverse the ASTs.
+
+```
+data _⊢̂_⊳̂_ (X : Set) : (X ⊢) → (X ⊢) → Set where
+--   refl : ∀ {X : Set} {x : X ⊢} → X ⊢̂ x ⊳̂ x
```

Comment:
Should this comment be removed?

Reply:
Yes ... unless it turns out that having a simple pattern for things that are identical is useful... It will certainly be quicker than needing to traverse subtrees that are completely identical.